### PR TITLE
feat: broken job not prevent the backup

### LINKED ISSUE #9: one broken job breaks all other jobs
Hello
We encountered an unpleasant situation: backups were not collected for a couple of days due to errors in the jobs config.
There are several jobs that worked properly, but among them there was one unconfigured job. As a result, all backups were not collected for several days, alerts also did not arrive by email, apparently because checking the correctness of backups did not work properly.

```
# nxs-backup --version
nxs-backup v3.0.2
```

after the “broken” config was removed from conf.d, everything works normally, but I would like to either receive alerts about a problem with the configs, or have live jobs processed separately, regardless of the presence of problematic jobs

### PR DESCRIPTION
**Pull request by randreev1321**
Change basic backup behavior:
- not prevent backups if there are problems during initialization of storages or jobs
- log and send a notification if there are errors during initialization 
Print errors and the list of backup jobs during configuration test. Update the nxs-go-appctx library to v3.
Use appctx routine to process commands.
Config file isn't need when selfupdate run.

Refs: #9

### CODE REVIEW

File: modules/cmd_handler/generate_config.go [hunk 1]
```diff
@@ -127,25 +125,40 @@ type smbParams struct {
...
+	done     chan error
+	cfgPath  string
+	jobType  string
+	outPath  string
+	arg      *arg.Parser
+	storages map[string]string
+}
 
-	var errs *multierror.Error
+func InitGenerateConfig(dc chan error, cfgPath, jt, op string, st map[string]string, ap *arg.Parser) *GenerateConfig {
```
Comment (borisershov) [inline modules/cmd_handler/generate_config.go]: Try to use the `Settings` struct to specify arguments to  module context `Init` function. Like [this](https://github.com/nixys/nxs-support-bot/blob/51a4429fdc90bced97bca9912a6a40f456c78512/modules/issues/issues.go#L12)

File: modules/cmd_handler/generate_config.go [hunk 2]
```diff
@@ -1,18 +1,16 @@
-package arg_cmd
+package cmd_handler
```
Comment (borisershov) [inline modules/cmd_handler/generate_config.go]: Try to use separate directories for each logical submodule. This way gives you an isolation and you may use same function and structure names in each module.

File: ctx/notifications_init.go [hunk 3]
```diff
@@ -7,8 +7,8 @@ import (
 	"github.com/hashicorp/go-multierror"
 	"github.com/sirupsen/logrus"
 
-	"nxs-backup/interfaces"
-	"nxs-backup/modules/backend/notifier"
+	"github.com/nixys/nxs-backup/interfaces"
+	"github.com/nixys/nxs-backup/modules/notifier"
 )
 
 var messageLevels = map[string]logrus.Level{
```
Comment (borisershov) [inline ctx/notifications_init.go:15]: Try to use case insensitive string compares to make this moment more optimal

File: ctx/context.go [hunk 4]
```diff
@@ -6,110 +6,158 @@ import (
...
+	Log       *logrus.Logger
+	Done      chan error
+	EventCh   chan logger.LogRecord
+	EventsWG  *sync.WaitGroup
+	Notifiers []interfaces.Notifier
 }
 
-// Init initiates application custom context
-func (c *Ctx) Init(opts appctx.CustomContextFuncOpts) (appctx.CfgData, error) {
+func AppCtxInit() (any, error) {
```
Comment (borisershov) [inline ctx/context.go]: Try to use `switch` statement to make this function more structural

**Review by borisershov — CHANGES_REQUESTED** [review]

File: modules/cmd_handler/generate_config.go (hunk 1 above)
Comment (randreev1321) [reply modules/cmd_handler/generate_config.go]: Used the `Opts` structure

File: modules/cmd_handler/generate_config.go (hunk 2 above)
Comment (randreev1321) [reply modules/cmd_handler/generate_config.go]: Done

File: ctx/notifications_init.go (hunk 3 above)
Comment (randreev1321) [reply ctx/notifications_init.go:15]: Used a conversion to uppercase

File: ctx/context.go (hunk 4 above)
Comment (randreev1321) [reply ctx/context.go]: done

**Review by borisershov — APPROVED** [review]

**Pull request merged**